packages/nextjs/src/config/withSentryConfig.ts

```diff
@@ -153,14 +153,57 @@ function getFinalConfigObject(
     }
   }
 
-  if (process.env.TURBOPACK && !process.env.SENTRY_SUPPRESS_TURBOPACK_WARNING) {
+  if (nextJsVersion) {
+    const { major, minor, patch, prerelease } = parseSemver(nextJsVersion);
+    const isSupportedVersion =
+      major !== undefined &&
+      minor !== undefined &&
+      patch !== undefined &&
+      (major > 15 ||
+        (major === 15 && minor > 3) ||
+        (major === 15 && minor === 3 && patch > 0 && prerelease === undefined));
+    const isSupportedCanary =
+      major !== undefined &&
+      minor !== undefined &&
+      patch !== undefined &&
+      prerelease !== undefined &&
+      major === 15 &&
+      minor === 3 &&
+      patch === 0 &&
+      prerelease.startsWith('canary.') &&
+      parseInt(prerelease.split('.')[1] || '', 10) >= 8;
+    const supportsClientInstrumentation = isSupportedCanary || isSupportedVersion;
+
+    if (supportsClientInstrumentation) {
+      incomingUserNextConfigObject.experimental = {
+        clientInstrumentationHook: true,
+        ...incomingUserNextConfigObject.experimental,
+      };
+    } else if (process.env.TURBOPACK) {
+      if (process.env.NODE_ENV === 'development') {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `[@sentry/nextjs] WARNING: You are using the Sentry SDK with Turbopack (\`next dev --turbo\`). The Sentry SDK is compatible with Turbopack on Next.js version 15.3.0 or later. You are currently on ${nextJsVersion}. Please upgrade to a newer Next.js version to use the Sentry SDK with Turbopack. Note that the SDK will continue to work for non-Turbopack production builds. This warning is only about dev-mode.`,
+        );
+      } else if (process.env.NODE_ENV === 'production') {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `[@sentry/nextjs] WARNING: You are using the Sentry SDK with Turbopack (\`next build --turbo\`). The Sentry SDK is compatible with Turbopack on Next.js version 15.3.0 or later. You are currently on ${nextJsVersion}. Please upgrade to a newer Next.js version to use the Sentry SDK with Turbopack. Note that as Turbopack is still experimental for production builds, some of the Sentry SDK features like source maps will not work. Follow this issue for progress on Sentry + Turbopack: https://github.com/getsentry/sentry-javascript/issues/8105.`,
+        );
+      }
+    }
+  } else {
+    // If we cannot detect a Next.js version for whatever reason, the sensible default is still to set the `experimental.instrumentationHook`.
+    incomingUserNextConfigObject.experimental = {
+      clientInstrumentationHook: true,
+      ...incomingUserNextConfigObject.experimental,
+    };
```

Comment:
It seems like setting this option even though it does not exist yet does not cause any warnings for whatever reason. This seems to be different from other experiments.